ruby/rspec/tutorial.md

```diff
@@ -0,0 +1,155 @@
+# RSpec Tutorial/Guide
+
+RSpec is a testing tool for Ruby, it is the most frequently used testing library in Ruby applications, where other tools in other languages were inspired by like jasmine and mocha.  This tutorial will aim to steer you into the path of testing and show you the different ways of testing.
+
+<img src="/ruby/rspec/levels.jpg" width="420px" height="500px"/>
+
+There are some key parts in a RSpec test:
+* `describe`: it is a example group, usually groups the tests of a method or class.
+* `it`: the atom of the tests, it should contain one test/expectation.
+* `before`: hook that executer before a  group of tests, depending where is located, if it’s located inside a describe or context, it will be executed before  the tests inside the describe or context. By default runs before each test, but you can pass the option `:all` to run before all of them only.  It’s usually used for testing setup
+* `after`: same as before, but after the tests, it’s usually used for teardown setup.
+* `context`: same behavior than `describe` but it’s used for a semantic purpose: each block of tests inside the context behaves in a certain when or under certain conditions.
```

Comment:
`same behaviour as`

`in a certain way`

Reply:
behavior is the American one, yours is the British, like colour